staging/src/k8s.io/apiserver/pkg/admission/plugin/namespace/lifecycle/admission.go

```diff
@@ -91,17 +91,20 @@ func (l *lifecycle) Admit(a admission.Attributes) error {
 		return errors.NewForbidden(a.GetResource().GroupResource(), a.GetName(), fmt.Errorf("this namespace may not be deleted"))
 	}
 
-	// if we're here, then we've already passed authentication, so we're allowed to do what we're trying to do
-	// if we're here, then the API server has found a route, which means that if we have a non-empty namespace
-	// its a namespaced resource.
-	if len(a.GetNamespace()) == 0 || a.GetKind().GroupKind() == v1.SchemeGroupVersion.WithKind("Namespace").GroupKind() {
+	// always allow non-namespaced resources
+	if len(a.GetNamespace()) == 0 {
```

Comment:
I can never remember if a namespace resource has a namespace set at all points in the stack.  Mind adding a `&& kind != "namespace"` sort of check here?